upgrade_test.py

```diff
@@ -29,39 +29,33 @@
 from sdcm.cluster import BaseNode
 from sdcm.fill_db_data import FillDatabaseData
 from sdcm.stress_thread import CassandraStressThread
-from sdcm.utils.version_utils import is_enterprise, get_node_supported_sstable_versions
+from sdcm.utils.version_utils import get_node_supported_sstable_versions
 from sdcm.sct_events.system import InfoEvent
 from sdcm.sct_events.database import IndexSpecialColumnErrorEvent
 from sdcm.sct_events.group_common_events import ignore_upgrade_schema_errors, ignore_ycsb_connection_refused, \
     ignore_abort_requested_errors, decorate_with_context
 from sdcm.utils import loader_utils
-from sdcm.utils.version_utils import parse_scylla_version
 from test_lib.sla import create_sla_auth
 
 
+NUMBER_OF_ROWS_FOR_TRUNCATE_TEST = 10
+
+
 def truncate_entries(func):
     @wraps(func)
     def inner(self, *args, **kwargs):
-        # Perform validation of truncate entries in case the new version is 3.1 or more
         node = args[0]
-        if self.truncate_entries_flag:
-            base_version = self.params.get('scylla_version')
-            system_truncated = bool(parse_scylla_version(base_version) >= parse_scylla_version('3.1')
-                                    and not is_enterprise(base_version))
-            with self.db_cluster.cql_connection_patient(node, keyspace='truncate_ks') as session:
-                self.cql_truncate_simple_tables(session=session, rows=self.insert_rows)
-                self.validate_truncated_entries_for_table(session=session, system_truncated=system_truncated)
+        with self.db_cluster.cql_connection_patient(node, keyspace='truncate_ks') as session:
+            self.cql_truncate_simple_tables(session=session, rows=NUMBER_OF_ROWS_FOR_TRUNCATE_TEST)
+            self.validate_truncated_entries_for_table(session=session, system_truncated=True)
 
         func_result = func(self, *args, **kwargs)
 
-        result = node.remoter.run('scylla --version')
-        new_version = result.stdout
-        if new_version and parse_scylla_version(new_version) >= parse_scylla_version('3.1'):
-            # re-new connection
-            with self.db_cluster.cql_connection_patient(node, keyspace='truncate_ks') as session:
-                self.validate_truncated_entries_for_table(session=session, system_truncated=True)
-                self.read_data_from_truncated_tables(session=session)
-                self.cql_insert_data_to_simple_tables(session=session, rows=self.insert_rows)
+        # re-new connection
+        with self.db_cluster.cql_connection_patient(node, keyspace='truncate_ks') as session:
+            self.validate_truncated_entries_for_table(session=session, system_truncated=True)
+            self.read_data_from_truncated_tables(session=session)
+            self.cql_insert_data_to_simple_tables(session=session, rows=NUMBER_OF_ROWS_FOR_TRUNCATE_TEST)
```

Comment:
why do we do this twice?

Reply:
I'm sure you're refering to the `self.validate_truncated_entries_for_table(session=session, system_truncated=True)` call. The whole function is used as a decorator that is used on top of `upgrade_node` and `rollback_node`, so we call `validate_truncated_entries_for_table` before and after the function call to make sure that the system tables were unaltered by the upgrading/rollbacking process. as seen here: https://github.com/scylladb/scylla-cluster-tests/pull/1042